R/runfiles_commands.bzl

```diff
@@ -0,0 +1,58 @@
+# Copyright 2017 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+_RUNFILES_CMDS = """
```

Comment:
This is loosely based on: https://github.com/bazelbuild/rules_nodejs/blob/master/internal/node/node_launcher.sh

I use the template because this is the only one I found that work in all cases:
1) When you use bazel run
2) When you use bazel test
3) When you invoke the executable directly
4) When you use the binary as part of the runfiles of another binary
5) When you use both the sandboxed and standalone mode

But maybe the solution adopted by the docker rules work in all those cases as well, I'll have to check, because the symlinks in Bazel are everywhere and this kind of things can get really entangled.

Reply:
I will take care of it later. Thank you for explaining the origin.